lib/widgets/chatbot/core/chat_window.dart

```diff
@@ -211,13 +210,44 @@ class _InteractiveChatWindow extends State<InteractiveChatWindow> {
   }
 
   void setFeedbackView(int detail) async {
-    Provider.of<ChatProvider>(context, listen: false).feedbackDetail(-1, detail);
+    Provider.of<ChatProvider>(context, listen: false)
+        .feedbackDetail(-1, detail);
     if (detail != -1) await Future.delayed(const Duration(milliseconds: 300));
     setState(() {
       _feedbackOpen = !_feedbackOpen;
     });
   }
 
+  /* in "prompt" mode, this method adds a prompt from the bot into 
+     the conversation */
+  void newPrompt() async {
+    if (Provider.of<ConfigProvider>(context, listen: false).getMode() ==
+        Mode.prompt) {
+      Provider.of<ChatProvider>(context, listen: false)
+          .addBotResponse(MessagingStrings.demoPrompts[promptNum++], false);
+      // loop back to the first prompt
+      if (promptNum >= MessagingStrings.demoPrompts.length) {
+        promptNum = 0;
+      }
+
+      Map<String, Object>? botMessage =
+          Provider.of<ChatProvider>(context, listen: false).getBotMessage();
+
+      if (botMessage != null) {
+        botMessage['convoID'] = convoID;
+        FirebaseDbService.addMessageData(botMessage);
+
+        channel.sink.add('{"text": "[DONE]"}');
+
+        // clears bubble list before the conversation is re filled to include
+        // the prompt now
+        Provider.of<ChatProvider>(context, listen: false).restartConvo();
+      }
```

Comment:
I don't think this this effectively restarts the conversation with the new prompt. The prompt does not effect the conversation at all. Like when I continue talking with it, it will not refer to anything in the prompt. I am honestly not sure of the benefit of having a "new prompt" button. I sent a message to Nick to get his thoughts about the prompt feature as a whole.